aioamqp/tests/test_publish.py

```diff
@@ -51,3 +52,28 @@ def test_confirmed_publish(self):
         queues = self.list_queues()
         self.assertIn("q", queues)
         self.assertEqual(1, queues["q"]['messages'])
+
+    @testing.coroutine
+    def test_return_from_publish(self):
+        called = False
+
+        @asyncio.coroutine
+        def callback(channel, body, envelope, properties):
+            nonlocal called
+            called = True
+
+        self.channel.return_callback = callback
```

Comment:
you should create a new channel, with the parameter in the init() or in the factory `self.amqp.create_channel(callback=callback)` IIRC